### PR TITLE
fix(ingest/s3): Converting windows style path to posix one on local fs

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/s3/source.py [hunk 1]
```diff
@@ -819,7 +820,10 @@ def local_browser(self, path_spec: PathSpec) -> Iterable[Tuple[str, datetime, in
                 dirs.sort(key=functools.cmp_to_key(partitioned_folder_comparator))
 
                 for file in sorted(files):
-                    full_path = os.path.join(root, file)
+                    # We need to make sure the path is in posix style which is not true on windows
+                    full_path = PurePath(
+                        os.path.normpath(os.path.join(root, file))
+                    ).as_posix()
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/s3/source.py:826]: Do we need both os.path.normpath and PurePath.as_posix ?

From docs, both do similar things.
https://docs.python.org/3/library/os.path.html#os.path.normpath
https://docs.python.org/3/library/pathlib.html

Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/s3/source.py:826]: If I get it right doesn't do the same thing even though there are similarities:
1. normpath -> 
```
os.path.normpath() method in Python is used to normalize the specified path. All redundant separator and up-level references are collapsed in the process of path normalization. 

For example: A//B, A/B/, A/./B and A/foo/../B all will be normalized to A/B. 

On Windows operating system, any forward slash (‘/’) in the path is converted to backslash (‘\’).
```
This is needed to make the dir separator consistent and also it simplifies the path (which I think is a good thing)
2. PurePath's as_posix turns the path to a UNIX path and uses unix type dir separators which is what s3 source and the urn expects

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/s3/source.py:826]: right they aren't same. Sounds like normpath does something more than as_posix, but yet it already does everything that as_posix might be doing ? (linux like  / separators)

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/s3/source.py:826]: Didn't understand the value add of PurePath.as_posix over os.path.normpath but the change seems better than original code.

**Review by mayurinehate — APPROVED** [review]
lgtm

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**